### DISCUSSION TITLE
[QUESTION] Any way to remove a limitation for HLS not generating if there was no data last 30 seconds?

### DISCUSSION THREAD

**Discussion by sam-vasiutin** [awslabs/amazon-kinesis-video-streams-producer-c#364] [Q&A] [answered]
I am using kinesis for building live-streaming iOS application. In live there are situations when streamer's device doesn't pass any data for 30 seconds or later (conditionally, paused state of stream). 

For playback I am generating HLS url and playing it back with AVPlayer, and because of this situations, when in the middle of stream can be no data available for 30 or more seconds this url fails to continue playing.

So I have had to build an additional logic for updating HLS url, **but is there any way to not to break current one**?

https://docs.aws.amazon.com/kinesisvideostreams/latest/dg/API_reader_GetHLSStreamingSessionURL.html

```
ResourceNotFoundException
GetImages will throw this error when Kinesis Video Streams can't find the stream that you specified.

GetHLSStreamingSessionURL and GetDASHStreamingSessionURL throw this error if a session with a PlaybackMode of ON_DEMAND or LIVE_REPLAYis requested for a stream that has no fragments within the requested time range, or if a session with a PlaybackMode of LIVE is requested for a stream that has no fragments within the last 30 seconds.

HTTP Status Code: 404
```

**It says HLS url will not be generated for a stream that has no fragments within the last 30 seconds, but it also breaks the existing HLS urls, that are already playing back a live stream.**

When the stream is paused for 0-29 seconds in live, AVPlayer stalls for the time while there is no data and just continues when it is available.

**Comment by jdelapla** [comment, marked as answer]
I've asked the playback team, and unfortunately the answer is no. This is part of the API design.
